src/tests/ftest/soak/soak.py

```diff
@@ -20,287 +20,433 @@
-sys.path.append('./../../utils/py')
-import server_utils
-import write_host_file
-import ior_utils
+from apricot import TestWithServers
+from ior_utils import IorCommand
 import slurm_utils
-import dmg_utils
-from daos_api import DaosContext, DaosPool, DaosApiError
+from general_utils import TestPool
+from ClusterShell.NodeSet import NodeSet
+from avocado.utils import process
+import socket
 
 
-class Soak(Test):
-    """
-    Test class Description: DAOS Soak test cases
+class Soak(TestWithServers):
+    """Execute DAOS Soak test cases.
+
+    :avocado: recursive
+    Args:
+        TestWithServers (AvocadoTest): Unit Test test cases
+    There are currently two types of soak tests.
+        1) smoke - runs each specified cmdline (job spec) for a single
+           iteration. The smoke test is to verify the environment is
+           configured properly before running the longer soaks
+        2) 1 hour - this will run a defined set of jobs and continue to submit
+           the jobs until the time has expired.
+
+    Soak Test will require a Slurm partition that contains the client nodes
+    that are listed in the yaml file.
+
+    The tests also use an IOR that is compiled with MPICH and is built with
+    both the DAOS and MPI-IO drivers.
+
     """
 
     def job_done(self, args):
-        """
-        This is a callback function called when a job is done
+        """Call this function when a job is done.
 
-        handle --which job, i.e. the job ID
-        state  --string indicating job completion status
+        Args:
+            args (list):handle --which job, i.e. the job ID,
+                        state  --string indicating job completion status
         """
-
         self.soak_results[args["handle"]] = args["state"]
 
+    def create_pool(self, path):
+        """Create a pool that the various tests use for storage.
 
-    def create_pool(self):
-        """
-        Creates a pool that the various tests use for storage.
-        """
+        Args:
+            path: pool params from yaml file  /run/pool/* is default
+        Returns:
+            object: TestPool object
 
-        createmode = self.params.get("mode", '/run/pool1/createmode/*/')
-        createuid = os.geteuid()
-        creategid = os.getegid()
-        createsetid = self.params.get("setname", '/run/pool1/createset/')
-        createsize = self.params.get("size", '/run/pool1/createsize/')
-        self.createsvc = self.params.get("svcn", '/run/pool1/createsvc/')
+        """
+        # Create a pool
+        pool = TestPool(self.context, self.log)
+        pool.get_params(self, path)
+        pool.create()
+        self.log.info("Valid Pool UUID is %s", pool.uuid)
+
+        # Check that the pool was created
+        self.assertTrue(
+            pool.check_files(self.hostlist_servers),
+            "Pool data not detected on servers")
+        return pool
+
+    def destroy_pool(self, pool):
+        """Destroy the specified pool - TO DO."""
+        pass
```

Comment:
Any specific reason this needs to be still done?
Can't we use destroy method in TestPool to implement it?